neqo-transport/src/cc/tests/cubic.rs

```diff
@@ -0,0 +1,301 @@
+    let mut next_pn_send = 0;
+    let mut next_pn_ack = 0;
+
+    next_pn_send = fill_cwnd(&mut cubic, next_pn_send, now);
+
+    // This will start with TCP phase.
+    // in this phase cwnd is increase by CUBIC_ALPHA every RTT. We can look at it as
+    // increase of MAX_DATAGRAM_SIZE every 1 / CUBIC_ALPHA RTTs.
+    // The phase will end when cwnd calculated with cubic equation is equal to TCP estimate:
+    // CUBIC_C * (n * RTT / CUBIC_ALPHA)^3 * MAX_DATAGRAM_SIZE = n * MAX_DATAGRAM_SIZE
+    // from this n = sqrt(CUBIC_ALPHA^3/ (CUBIC_C * RTT^3)).
+    let num_tcp_increases = (CUBIC_ALPHA.powi(3) / (CUBIC_C * RTT.as_secs_f64().powi(3)))
+        .sqrt()
+        .round() as u64;
+
+    for _ in 0..num_tcp_increases {
+        let cwnd_rtt_start = cubic.cwnd();
+        //Expected acks during a period of RTT / CUBIC_ALPHA.
+        let acks = (f64::try_from(i32::try_from(cwnd_rtt_start).unwrap()).unwrap()
+            / MAX_DATAGRAM_SIZE_F64
+            / CUBIC_ALPHA)
+            .round() as u64;
+        // The time between acks if they are ideally paced over a RTT.
+        let time_increase = RTT / u32::try_from(cwnd_rtt_start / MAX_DATAGRAM_SIZE).unwrap();
+
+        for _ in 0..acks {
+            ack_packet(&mut cubic, next_pn_ack, now);
+            next_pn_ack += 1;
+            next_pn_send = fill_cwnd(&mut cubic, next_pn_send, now);
+            now += time_increase;
+        }
+
+        assert_eq!(cubic.cwnd() - cwnd_rtt_start, MAX_DATAGRAM_SIZE);
+    }
+
+    // The next increase will be according to the cubic equation.
+
+    let cwnd_rtt_start = cubic.cwnd();
+    // calculate new time_increase.
+    let time_increase = RTT / u32::try_from(cwnd_rtt_start / MAX_DATAGRAM_SIZE).unwrap();
+    let mut num_acks = 0; // count the number of acks. until cwnd is increased by MAX_DATAGRAM_SIZE.
+
+    while cwnd_rtt_start == cubic.cwnd() {
+        num_acks += 1;
+        ack_packet(&mut cubic, next_pn_ack, now);
+        next_pn_ack += 1;
+        next_pn_send = fill_cwnd(&mut cubic, next_pn_send, now);
+        now += time_increase;
+    }
+
+    // Make sure that the increase is not according to TCP equation, i.e tha it took
+    // less than RTT / CUBIC_ALPHA.
+    let expected_ack_tcp_increase = (f64::try_from(u32::try_from(cwnd_rtt_start).unwrap()).unwrap()
+        / MAX_DATAGRAM_SIZE_F64
+        / CUBIC_ALPHA)
+        .round() as u64;
+    assert!(num_acks < expected_ack_tcp_increase);
+
+    // This first increase after a TCP phase may be shorter than what it would take by a regular cubic phase,
+    // because of the proper byte counting and the credit it already had before entering this phase. Therefore
+    // We will perform another round and compare it to expected increase using the cubic equation.
+
+    let cwnd_rtt_start_after_tcp = cubic.cwnd();
+    let elapse_time = now - start_time - time_increase; // "- time_increase" because it is added above.
```

Comment:
As below, you can avoid the extra subtraction by increasing now before calling ack_packet/fill_cwnd.  That also avoids having the first ACK being right on top of the packet being sent.

Also "elapsed_time".